kubernetes/doc/create-options.md

```diff
@@ -0,0 +1,122 @@
+# Kubernetes createOptions Extensions
+
+IoT Edge on Kubernetes supports several experimental features to allow a user to assign additional parameters to pods that represents IoT Edge Modules.
+
+## Experimental features
+
+All features described in this document are experimental. To enable them, the following environment variable need to be set for the edgeAgent (make note of the double underscores):
+
+| Environment Variable Name                   | value  |
+|---------------------------------------------|--------|
+| `ExperimentalFeatures__Enabled`             | `true` |
+| `ExperimentalFeatures__EnableK8SExtensions` | `true` |
+
+## Create Options
+
+We added CreateOptions for experimental features on Kubernetes. These options will be ignores until both `ExperimentalFeatures__Enabled` and `ExperimentalFeatures__Enabled{FeatureName}` are not set to `true`.
+
+```json
+{
+  "k8s-experimental": {
+    "volumes": [{...}],
+    "resources": [{...}],
+    "nodeSelector": {...}
+  }
+}
+```
+
+## Volumes
+
+EdgeAgent allows to mount existing volumes in the namespace from different sources e.g. `ConfigMap` as a pod `Volume`.
+
+### Create Options
+
+A `volumes` section of config used to describe how a `ConfigMap` or any other volume source existing in a namespace will be mounted into module container. The value of this config section is an array of pairs `volume` and `volumeMount`. A `volume` part describes how a specified volume source is mounted to `pod` and `volumeMounts` describes a mounting of a `volume` within a module main container. 
```

Comment:
Maybe add at the end, 
"See [Volumes](https://kubernetes.io/docs/concepts/storage/volumes/) and [PodSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#podspec-v1-core) reference for more information."

Reply:
Here or at the end of the 2nd paragraph....